tests/providers/cncf/kubernetes/executors/test_kubernetes_executor.py

```diff
@@ -1419,7 +1445,128 @@ def test_clear_not_launched_queued_tasks_not_launched_other_queue(
         assert mock_kube_client.list_namespaced_pod.call_count == 0
 
     @pytest.mark.db_test
+    @pytest.mark.skipif(
+        not hasattr(TaskInstance, "executor"), reason="Hybrid executor added in later version"
+    )
     @mock.patch("airflow.providers.cncf.kubernetes.executors.kubernetes_executor.DynamicClient")
+    @conf_vars({("core", "executor"): KUBERNETES_EXECUTOR})
+    def test_clear_not_launched_queued_tasks_not_launched_other_executor(
+        self, mock_kube_dynamic_client, dag_maker, create_dummy_dag, session
+    ):
+        """Queued TI has no pod, but it is not queued for the k8s executor"""
+        mock_kube_client = mock.MagicMock()
+        mock_kube_dynamic_client.return_value = mock.MagicMock()
+        mock_pod_resource = mock.MagicMock()
+        mock_kube_dynamic_client.return_value.resources.get.return_value = mock_pod_resource
+        mock_kube_dynamic_client.return_value.get.return_value.items = []
+
+        # This is hack to use overridden conf vars as it seems executors loaded before conf override.
+        if hasattr(TaskInstance, "executor"):
+            from airflow.executors.executor_loader import ExecutorLoader
+
+            ExecutorLoader.get_executor_names().clear()
+        create_dummy_dag(dag_id="test_clear", task_id="task1", with_dagrun_type=None)
+        dag_run = dag_maker.create_dagrun()
+
+        ti = dag_run.task_instances[0]
+        ti.state = State.QUEUED
+        ti.queued_by_job_id = 1
+        ti.executor = "CeleryExecutor"
+        session.flush()
+
+        executor = self.kubernetes_executor
+        executor.job_id = 1
+
+        executor.kube_client = mock_kube_client
+        executor.clear_not_launched_queued_tasks(session=session)
+
+        ti.refresh_from_db()
+        assert ti.executor == "CeleryExecutor"
```

Comment:
Can you also add a test case for when there is default executor that is not k8s and we should expect no change to state as well.

Reply:
I think test_clear_not_launched_queued_tasks_not_launched_other_executor and test_clear_not_launched_queued_tasks_launched_none_executor covers that

Reply:
Both those cases have k8s as the default executor though? I don't see how those cover the case where k8s is _not_ the default executor.

Reply:
added